storage/remote/otlptranslator/prometheus/normalize_label.go

```diff
@@ -28,14 +30,15 @@ import (
 // Labels that start with non-letter rune will be prefixed with "key_".
 //
 // An exception is made for double-underscores which are allowed.
-func NormalizeLabel(label string) string {
+func NormalizeLabel(label string, allowUTF8 bool) string {
 	// Trivial case
 	if len(label) == 0 {
 		return label
 	}
 
-	// Replace all non-alphanumeric runes with underscores
-	label = strings.Map(sanitizeRune, label)
```

Comment:
I'm using `model.EscapeName` as a suggestion from @ywwg! 

The previous function isn't working as expected, it translates characters to underscores if `unicode.IsLetter(r) || unicode.IsDigit(r)` is not true.

This is problematic because:
- Foreign letters like russian, japanese, german complies with `unicode.IsLetter()` and they are being replaced when they shouldn't
- The character `:` is valid even without UTF8 and is being replaced. See https://github.com/prometheus/prometheus/issues/15199


It looks impractical to concatenate every foreign character in the sanitizeRunes conditional, so I'm just using the same function used in prometheus/common like owen suggested :)

Reply:
> I'm using model.EscapeName as a suggestion from @ywwg!

If we are making changes unrelated to the PR's topic (add un-escaped UTF8 mode), it should be a separate PR.

Reply:
https://github.com/prometheus/prometheus/pull/15249